### PR TITLE
adds mocha spec solution

### PR DESCRIPTION
**Pull request by mgabor18**

### CODE REVIEW

File: mocha/specs/test.spec.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  describe("Careers page", () => {
+    it("should be opened", () => {
+      return expect(element(by.css(".header__logo")).isDisplayed()).to.eventually.be.true;
+    });
+  });
+
+  describe("Search form", () => {
+    it("should be displayed", () => {
+      expect(element(by.css(".job-search__form")).isDisplayed()).to.eventually.be.true;
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js:34]: You use expect with return and without it inconsistently. Could you please use use it consistently.

File: mocha/specs/test.spec.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    });
+
+    describe("Location Filter Box", () => {
+      beforeEach(async () => {
+        await selectLocation();
+      });
+
+      it("should be able to select city and country", async () => {
+        expect(await element(by.css(".select2-selection__rendered")).getText()).to.equal(testData.city);
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Wouldn't like  you use chai-as-promise functionality in such cases? Could you please use it consistently.

File: mocha/specs/test.spec.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+      it("should have apply button for job", async () => {
+        expect(await element(by.css(`a.search-result__item-apply[href*='.${jobFormat}']`)).isDisplayed()).to.be.true;
+      });
+    });
+
+    describe("Applying to position", () => {
+      beforeEach(async () => {
+        await applyForJob();
+        browser.wait(ec.presenceOf(element(by.css(".form-component__description div"))), GLOBAL_TIMEOUT);
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: This wait could be in the applyForJob function. What do you think?

File: mocha/specs/test.spec.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  await element(by.cssContainingText(`a[href*='.${jobFormat}']`, "View and apply")).click();
+}
+
+async function searchJob() {
+  await selectLocation();
+  await selectDepartment();
+  await element(by.css(".recruiting-search__submit")).click();
+}
+
+async function selectLocation() {
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Haven't you considered, that you put the test data as attribute in the function?

File: mocha/specs/test.spec.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+  if ((await element(by.css(".select2-selection__rendered")).getText()) !== testData.city) {
+    await element(by.css(".select2-selection__arrow")).click();
+    browser.sleep(1000);
+    await element(by.css(`li[aria-label="${testData.country}"]`)).click();
+    await element(by.css(`[id*="${testData.city}"]`)).click();
+  }
+}
+
+async function selectDepartment() {
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Haven't you considered, that you put the test data as attribute in the function?

File: mocha/specs/test.spec.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+      });
+
+      it("should have displayed the name of the position", async () => {
+        const position = await element(by.css(".form-component__description div")).getText();
+        expect(position).to.equal(testData.positionName);
+      });
+
+      it("should have displayed the city of the position", async () => {
+        const location = await element(by.css(".form-component__location")).getText();
+        expect(location.includes(testData.country.toUpperCase())).to.be.true;
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Haven't you seen any assert to check that a string includes a  sub-string in chai? Could you please modify it in all cases.

File: mocha/specs/test.spec.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+      });
+    });
+
+    describe("Searching", () => {
+      beforeEach(async () => {
+        await searchJob();
+      });
+
+      it("should have proper job found ", async () => {
+        expect(await element(by.css(`a.search-result__item-name[href*='.${jobFormat}']`)).isDisplayed()).to.be.true;
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Could you please modify the parent locator to an ElementArrayFinder (They are elements.)? After that you can find the children locator in the first element of the ElementArrayFinder. Coud you please do that in all cases?

File: mocha/specs/test.spec.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+        expect(await location.includes(testData.country.toUpperCase())).to.be.true;
+      });
+
+      it("should have job with description", async () => {
+        let desc = await element(by.css(".search-result__item")).element(by.css(".search-result__item-description"));
+        expect(await desc.isDisplayed()).to.be.true;
+      });
+
+      it("should have apply button for job", async () => {
+        expect(await element(by.css(`a.search-result__item-apply[href*='.${jobFormat}']`)).isDisplayed()).to.be.true;
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js]: Could you please check the text of the button too?

File: mocha/specs/test.spec.js [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        expect(location.includes(testData.country.toUpperCase())).to.be.true;
+      });
+    });
+  });
+});
+
+async function load() {
+  await browser.get("https://www.epam.com/careers");
+  browser.wait(ec.elementToBeClickable(element(by.css(".header__logo"))), GLOBAL_TIMEOUT);
+  let isPresent = await browser.isElementPresent(by.css(".cookie-disclaimer__button"));
```
Comment (takacskalman1987) [inline mocha/specs/test.spec.js:117]: Could you please create a separate function for clicking on the cookie button? After that you can call that in this function.

**Review by takacskalman1987 — CHANGES_REQUESTED** [review]

File: mocha/specs/test.spec.js (hunk 1 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js:34]: resolved

File: mocha/specs/test.spec.js (hunk 9 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js:117]: made seperate function

File: mocha/specs/test.spec.js (hunk 2 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: made it consistent

File: mocha/specs/test.spec.js (hunk 3 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: moved it to proper function

File: mocha/specs/test.spec.js (hunk 4 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: done

File: mocha/specs/test.spec.js (hunk 5 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: done

File: mocha/specs/test.spec.js (hunk 6 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: modified to use proper chai assertion

File: mocha/specs/test.spec.js (hunk 8 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: checking the text

File: mocha/specs/test.spec.js (hunk 7 above)
Comment (mgabor18) [reply mocha/specs/test.spec.js]: Done

**Review by takacskalman1987 — APPROVED** [review]

**Pull request merged**